src/_pytest/logging.py

```diff
@@ -427,6 +457,22 @@ def __init__(self, config):
         if self._log_cli_enabled():
             self._setup_cli_logging()
 
+    def _create_formatter(self, log_format, log_date_format):
+        # color option doesn't exist if terminal plugin is disabled
+        color = getattr(self._config.option, "color", "no")
+        if color != "no" and ColoredLevelFormatter.LEVELNAME_FMT_REGEX.search(
+            log_format
+        ):
+            formatter = ColoredLevelFormatter(
+                create_terminal_writer(self._config), log_format, log_date_format
+            )
+        else:
+            formatter = logging.Formatter(log_format, log_date_format)
+
+        if not six.PY2:
+            formatter._style = PercentStyleMultiline(formatter._style._fmt)
```

Comment:
~~Can you write a comment here please explaining why this is needed only for Python 3? I didn't look deep into it, but it feels surprising that Python 3 does not support this.~~

I suggest to write your skip reason here then: `Formatter classes don't support format styles in PY2` 😁